ui/app/scripts/controllers/newDeploymentCtrl.js

```diff
@@ -303,40 +344,168 @@ angular.module('apollo')
+                          var index = $scope.selectedServices.indexOf(service);
+                          if (index > -1) {
+                            $scope.selectedServices.splice(index, 1);
+                          }
+                          var service = allServices.find(x => x.id === serviceId);
+                          var serviceName = service.name;
+                          servicesNames = servicesNames.concat(serviceName);
+                          if(($scope.selectedServices).includes(service) == false) {
+                              var index = $scope.selectedServicesAndStacks.indexOf(stack);
+                              toggleSelectedService(service, index);
+                          }
+                        });
+                    }
+                }
+                else {
+                    var index = $scope.selectedServices.indexOf(service);
+                    if (index > -1) {
+                        $scope.selectedServices.splice(index, 1);
+                    }
+                    $scope.selectedServicesAndStacks.push(service);
+                    var index = $scope.selectedServicesAndStacks.indexOf(service);
+                    toggleSelectedService(service, index);
+                }
+            }
+            $scope.servicesNamesOfStacks[service.id] = servicesNames;
+            updateServicesNames();
+        };
+
+        function toggleSelectedService(service, index) {
+            if($scope.isGroupDeployment == false) {
                 if (service.isPartOfGroup) {
-                    $scope.selectedServices = [];
-                    deploymentSteps = ["choose-service", "choose-environment", "choose-groups", "choose-version", "confirmation"];
-                    $scope.selectedServices.push(service);
-                    $scope.isGroupDeployment = true;
+                    if($scope.selectedServices.length == 0) {
+                        $scope.selectedServices = [];
+                        deploymentSteps = ["choose-service", "choose-environment", "choose-groups", "choose-version", "confirmation"];
+                        $scope.selectedServices.push(service);
+                        $scope.isGroupDeployment = true;
+                    }
+                    else {
+                        $scope.selectedServicesAndStacks.splice(index, 1);
+                    }
+
                 } else {
                     $scope.selectedServices = $scope.selectedServices.filter(function(a){return !a.isPartOfGroup});
                     $scope.selectedServices.push(service);
-                    $scope.isGroupDeployment = false;
                     deploymentSteps = ["choose-service", "choose-environment", "choose-version", "confirmation"];
                 }
             }
-            updateServicesNames();
+            else {
+                if($scope.selectedServices.length == 0) {
+                    $scope.isGroupDeployment = false;
```

Comment:
What? why?
What is the purpose of the previous "if" section? I don't understand this change